pkg/progress/action_single_phase_test.go

```diff
@@ -0,0 +1,197 @@
+		context.Background(),
+		s.actionSet.GetName(),
+		metav1.DeleteOptions{})
+	c.Assert(actionSetErr, IsNil)
+}
+
+func (s *TestSuiteSinglePhase) createFixtures(blueprint *crv1alpha1.Blueprint, actionSet *crv1alpha1.ActionSet) error {
+	createdBlueprint, err := s.clientset.CrV1alpha1().Blueprints(blueprint.GetNamespace()).Create(
+		context.Background(),
+		blueprint,
+		metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	s.blueprint = createdBlueprint
+
+	createdActionSet, err := s.clientset.CrV1alpha1().ActionSets(actionSet.GetNamespace()).Create(
+		context.Background(),
+		actionSet,
+		metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	s.actionSet = createdActionSet
+
+	return nil
+}
+
+func (s *TestSuiteSinglePhase) TestUpdateActionsProgress(c *C) {
+	now := time.Now()
+	var testCases = []struct {
+		indexAction int
+		indexPhase  int
+		phaseState  crv1alpha1.State
+		expected    string
+	}{
+		{
+			phaseState: crv1alpha1.StatePending,
+			expected:   progressPercentStarted,
+		},
+		{
+			phaseState: crv1alpha1.StateRunning,
+			expected:   progressPercentStarted,
+		},
+		{
+			phaseState: crv1alpha1.StateFailed,
+			expected:   progressPercentStarted,
+		},
+		{
+			phaseState: crv1alpha1.StateComplete,
+			expected:   progressPercentCompleted,
+		},
+	}
+
+	for id, tc := range testCases {
+		assertActionProgress(
+			c,
+			s.clientset,
+			s.actionSet,
+			now,
+			tc.indexAction,
+			tc.indexPhase,
+			tc.phaseState,
+			tc.expected,
+			id)
+	}
+}
+
+func assertActionProgress(
+	c *C,
+	clientset versioned.Interface,
+	actionSet *crv1alpha1.ActionSet,
+	now time.Time,
+	indexAction int,
+	indexPhase int,
+	newState crv1alpha1.State,
+	expectedPercentCompleted string,
+	testCaseID int) {
+	// update the phase state
+	actionSet.Status.Actions[indexAction].Phases[indexPhase].State = newState
+	updated, err := clientset.CrV1alpha1().ActionSets(actionSet.GetNamespace()).Update(context.Background(), actionSet, metav1.UpdateOptions{})
+	c.Assert(err, IsNil)
+
+	// calculate and update the progress so that it reflects the state change
+	phaseWeights, totalWeight, err := calculatePhaseWeights(context.Background(), actionSet.GetName(), actionSet.GetNamespace(), clientset)
+	c.Assert(err, IsNil)
+
+	updateErr := updateActionsProgress(context.Background(), clientset, updated, phaseWeights, totalWeight, now)
+	c.Assert(updateErr, IsNil)
+
+	// retrieve the latest actionSet resource to confirm its progress data
+	actual, err := clientset.CrV1alpha1().ActionSets(actionSet.GetNamespace()).Get(context.Background(), updated.GetName(), metav1.GetOptions{})
+	c.Assert(err, IsNil, Commentf("test case #: %d", testCaseID))
+	c.Assert(actual.Status.Progress.PercentCompleted, Equals, expectedPercentCompleted, Commentf("test case #: %d", testCaseID))
+	c.Assert(actual.Status.Progress.LastTransitionTime.Time, Equals, now, Commentf("test case #: %d", testCaseID))
```

Comment:
```suggestion
	c.Assert(actual.Status.Progress.LastTransitionTime, NotNil)
	c.Assert(actual.Status.Progress.LastTransitionTime.Time, Equals, now, Commentf("test case #: %d", testCaseID))
```
We should probably add a non-nil check here